src/NuGet.Core/NuGet.Build.Tasks/StaticGraphRestoreTaskBase.cs

```diff
@@ -97,34 +103,81 @@ public override bool Execute()
 #endif
                 MSBuildLogger logger = new MSBuildLogger(Log);
 
+                Dictionary<string, string> globalProperties = GetGlobalProperties();
+
                 using (var semaphore = new SemaphoreSlim(initialCount: 0, maxCount: 1))
                 using (var loggingQueue = new TaskLoggingQueue(Log))
                 using (var process = new Process())
                 {
+                    bool errorLogged = false;
+
                     process.EnableRaisingEvents = true;
                     process.StartInfo = new ProcessStartInfo
                     {
-                        Arguments = $"\"{string.Join("\" \"", GetCommandLineArguments())}\"",
+                        Arguments = GetCommandLineArguments(globalProperties),
                         CreateNoWindow = true,
                         FileName = GetProcessFileName(ProcessFileName),
+                        RedirectStandardError = true,
                         RedirectStandardInput = true,
                         RedirectStandardOutput = true,
+#if !NETFRAMEWORK
+                        StandardInputEncoding = new UTF8Encoding(encoderShouldEmitUTF8Identifier: false),
```

Comment:
``` suggestion
                        StandardInputEncoding = Encoding.UTF8;
```
If we have to use the constructor, how about using the other constructor that accepts `throwOnInvalidBytes` boolean. [Docs](https://learn.microsoft.com/dotnet/api/system.text.utf8encoding.-ctor?view=net-7.0#system-text-utf8encoding-ctor(system-boolean-system-boolean))suggest that `For security reasons, you should enable error detection by calling a constructor that includes a throwOnInvalidBytes parameter and setting its value to true.`

Reply:
Done